src/Utilities/CssConcatenator/RuleBlock.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Utilities\CssConcatenator;
+
+/**
+ * This class represents a CSS rule block.
+ *
+ * @internal
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ * @author SignpostMarv
+ * @author Oliver Klee <github@oliverklee.de>
+ */
+class RuleBlock
+{
+    /**
+     * array whose keys are selectors for the rule block (values are of no significance)
+     *
+     * @var array
+     *
+     * @psalm-var array<string, mixed>
+     */
+    private $selectorsAsKeys = [];
+
+    /**
+     * the property declarations, e.g. "margin-top: 0.5em; padding: 0"
+     *
+     * @var string
+     */
+    private $declarationsBlock;
+
+    /**
+     * @param array $selectorsAsKeys
+     * @param string $declarationsBlock
+     *
+     * @psalm-param array<string, mixed> $selectorsAsKeys
+     */
+    public function __construct(array $selectorsAsKeys, string $declarationsBlock)
```

Comment:
@JakeQZ, @SignpostMarv Does is make sense that an empty array and an empty string can be passed, or would this indicate a problem upstream in the data flow, and an throwing an exception would be in order?

Reply:
do variants on `;\n;` create a declaration with an empty string ?

Reply:
If we choose to throw an exception in either case we should catch it in `CssInliner` if not in 'debug' mode, convert it to a `trigger_error`, and then carry on (the same as what we do when we get an exception from `CssSelectorConverter`).

I think the selectors array will necessarily be nonempty because I don't think CSS like `/* no selector */ { color: red; }` will get through to here.  But I'm less certain what happens with CSS like `body { /* non declarations */ }`

Reply:
> convert it to a `trigger_error`

Checking this, we actually silently ignore the exceptions if not in debug mode.  And with good reason I think - `trigger_error` could clutter up error logs if there's an upstream issue in a third-party component that can't be easily resolved.  (There might be a justification for three 'debug' options, with an additional one that does a `trigger_error`, but that's another matter.)

Reply:
Make the constructor non-public and validate before it hits the constructor?